File: lib/src/models/package_info.dart
import 'package:json_annotation/json_annotation.dart';

part 'package_info.g.dart';

@JsonSerializable()
class PackageInfo {
  const PackageInfo({
    required this.packageName,
    this.appIcon,
    this.version,
    this.appName,
    this.installationSource,
  });

  final String packageName;
  final String? appIcon;
  final String? appName;
  final String? version;
  final String? installationSource;

  factory PackageInfo.fromJson(Map<String, dynamic> json) =>
      _$PackageInfoFromJson(json);
}

File: android/src/main/kotlin/com/aheaditec/freerasp/models/PackageInfo.kt
package com.aheaditec.freerasp.models

import org.json.JSONObject

data class PackageInfo(
    val packageName: String,
    val appIcon: String? = null,
    val appName: String? = null,
    val version: String? = null,
    val installationSource: String? = null
) {
    companion object {
        fun fromTalsec(packageInfo: android.content.pm.PackageInfo): PackageInfo {
            return PackageInfo(
                packageInfo.packageName,
                packageInfo.appIcon,
                packageInfo.appName,
                packageInfo.version,
                packageInfo.installationSource
            )
        }
    }

    fun toJson(): String {
        val json = JSONObject().put("packageName", packageName)
            .putOpt("appIcon", appIcon)
            .putOpt("appName", appName)
            .putOpt("version", version)
            .putOpt("installationSource", installationSource)

        return json.toString()
    }
}